packages/statemanager/src/ethersStateManager.ts

```diff
@@ -47,7 +47,7 @@ export class EthersStateManager implements StateManagerInterface {
       blockTag: BigInt(this.blockTag),
     })
     ;(newState as any).contractCache = new Map(this.contractCache)
-    ;(newState as any).storageCache = new Map(this.storageCache)
+    ;(newState as any).storageCache = this.storageCache
```

Comment:
This doesn't copy the storage cache while it was copied before, is this intended?

Reply:
Hmm, interesting point.  What should we do here?  The `DefaultStateManager` just creates a brand new cache using the same settings as the original (but not actually copying the cached values).  This is a bit of a different beast as the cache is the "actual" changed values.

Reply:
I think current copy semantics is to copy original cache, (we use vmcopy for e.g. to get a new copy but not with any uncommitted (to disk) modifications)

Reply:
Hmm, ok, I'll have to think how to do this.  Not sure exactly the best way to handle this.

Reply:
```suggestion
    ;(newState as any).storageCache = new StorageCache({ size: 10000, type: CacheType.LRU })
```

I would suggest to do this.